src/pytest_inmanta_lsm/remote_orchestrator.py

```diff
@@ -899,24 +914,73 @@ def wait_until_deployment_finishes(
-                response.result["model"]["done"],
-                response.result["model"]["total"],
-            )
-            return response.result["model"]["total"] - response.result["model"]["done"] <= 0
+        # Determin api version
+        response = self.client.get_version(self.environment, version)
+        assert response.result is not None
+        new_api = "done" not in response.result["model"]
 
-        retry_limited(is_deployment_finished, timeout)
-        result = self.client.get_version(self.environment, version)
-        assert result.result is not None
-        for resource in result.result["resources"]:
-            LOGGER.info(f"Resource Status:\n{resource['status']}\n{pformat(resource, width=140)}\n")
-            assert (
-                resource["status"] == desired_state
-            ), f"Resource status do not match the desired state, got {resource['status']} (expected {desired_state})"
+        if not new_api:
+
+            def is_deployment_finished() -> bool:
+                response = self.client.get_version(self.environment, version)
+                assert response.result is not None
+                LOGGER.info(
+                    "Deployed %s of %s resources",
+                    response.result["model"]["done"],
+                    response.result["model"]["total"],
+                )
+                return response.result["model"]["total"] - response.result["model"]["done"] <= 0
+
+            retry_limited(is_deployment_finished, timeout)
+            result = self.client.get_version(self.environment, version)
+            assert result.result is not None
+            for resource in result.result["resources"]:
+                LOGGER.info(f"Resource Status:\n{resource['status']}\n{pformat(resource, width=140)}\n")
+                assert (
+                    resource["status"] == desired_state
+                ), f"Resource status do not match the desired state, got {resource['status']} (expected {desired_state})"
+        else:
+            self.wait_for_released(version)
+
+            def get_key_numbers():
+                result = self.client.resource_list(self.environment, deploy_summary=True, limit=1)
+                assert result.code == 200
+                summary = result.result["metadata"]["deploy_summary"]
+                # {'by_state': {'available': 3, 'cancelled': 0, 'deployed': 12, 'deploying': 0, 'failed': 0, 'skipped': 0,
+                #               'skipped_for_undefined': 0, 'unavailable': 0, 'undefined': 0}, 'total': 15}
+                return (
+                    (
+                        summary["by_state"]["deployed"]
+                        + summary["by_state"]["failed"]
+                        + summary["by_state"]["skipped"]
+                        + summary["by_state"]["skipped_for_undefined"]
+                        + summary["by_state"]["unavailable"]
+                        + summary["by_state"]["undefined"]
```

Comment:
What about `cancelled`? Do we want to wait until the timeout to porentially see it? (not familiar with the meaning of that state).  I was wondering if we shouldn't go for `total - deploying - available` (feels like a more stable set of states as the one containing all the possible outcomes of a deployment)?

Reply:
this is what we use on all the core tests, ... 

i.e. it is all non-transient states (everything has reached an endpoint )

Reply:
Can we maybe reference this is in some way?

Reply:
I think that `inmanta.const.DONE_STATES` is what we want, but it is not part of the stable api, can it be added?